catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -150,9 +157,38 @@ public void init() {
   private CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
       throws UnsupportedQueryException, SourceUnavailableException, FederationException {
     QueryRequest request = cqlRequest.createQueryRequest(catalogFramework.getId(), filterBuilder);
-
     Stopwatch stopwatch = Stopwatch.createStarted();
-    QueryResponse response = catalogFramework.query(request);
+
+    List<QueryResponse> responses = Collections.synchronizedList(new ArrayList<>());
+    List<Result> results =
+        ResultIterable.resultIterable(
+                (queryRequest) -> {
+                  QueryResponse queryResponse = catalogFramework.query(queryRequest);
+                  responses.add(queryResponse);
+                  return queryResponse;
+                },
+                request,
+                cqlRequest.getCount())
+            .stream()
+            .collect(Collectors.toList());
+    QueryResponse response =
+        new QueryResponseImpl(
+            request,
+            results,
+            true,
+            responses
+                .stream()
+                .filter(Objects::nonNull)
+                .map(QueryResponse::getHits)
+                .findFirst()
+                .orElse(-1l),
+            responses
```

Comment:
✏️ This double-streaming feels very inefficient but nothing's jumping out at me to really improve it.